livesql/binlog.go

```diff
@@ -302,9 +312,26 @@ func (b *Binlog) parseBinlogRowsEvent(event *replication.BinlogEvent) (*update,
 	return update, nil
 }
 
+// SetUpdateDelay sets the duration by which future updates will be delayed.
+func (b *Binlog) SetUpdateDelay(d time.Duration) {
+	b.delayMu.Lock()
+	defer b.delayMu.Unlock()
+	b.delay = d
+}
+
 // RunPollLoop is the core binlog function that fetches and distributes updates
 // from MySQL
 func (b *Binlog) RunPollLoop() error {
+	updateCh := make(chan delayedUpdate, 1024)
```

Comment:
alternative magic number 1024 is creating unbound number of goroutines here.

Reply:
alternatively, the poll loop itself could sleep?

Reply:
poll loop shouldn't sleep because we don't want to wait replica lag between every binlog update